pubsub/iam.js

```diff
@@ -13,84 +13,48 @@
 
 'use strict';
 
-// [START auth]
+// [START setup]
 // By default, the client will authenticate using the service account file
 // specified by the GOOGLE_APPLICATION_CREDENTIALS environment variable and use
 // the project specified by the GCLOUD_PROJECT environment variable. See
 // https://googlecloudplatform.github.io/gcloud-node/#/docs/google-cloud/latest/guides/authentication
 var PubSub = require('@google-cloud/pubsub');
+// [END setup]
 
-// Instantiate a pubsub client
-var pubsub = PubSub();
-// [END auth]
-
-// [START get_topic_policy]
-/**
- * Retrieve a topic's IAM policy.
- *
- * @param {string} topicName The name of the topic.
- * @param {function} callback The callback function.
- */
 function getTopicPolicy (topicName, callback) {
-  if (!topicName) {
-    return callback(new Error('"topicName" is required!'));
-  }
-
-  // Grab a reference to an existing topic
+  var pubsub = PubSub();
   var topic = pubsub.topic(topicName);
 
   // Retrieve the IAM policy for the topic
-  topic.iam.getPolicy(function (err, policy) {
+  // See https://googlecloudplatform.github.io/google-cloud-node/#/docs/pubsub/latest/pubsub/topic?method=iam.getPolicy
+  topic.iam.getPolicy(function (err, policy, apiResponse) {
     if (err) {
       return callback(err);
     }
 
-    console.log('Got topic policy:', policy);
-    return callback(null, policy);
+    console.log('Got policy for topic: %s', topicName);
+    return callback(null, policy, apiResponse);
   });
 }
-// [END get_topic_policy]
 
-// [START get_subscription_policy]
```

Comment:
@jmdobry These are needed for https://cloud.google.com/pubsub/docs/access_control